app/models/flexible_criterion.rb

```diff
@@ -166,4 +166,9 @@ def set_mark_by_criterion(mark_to_change, mark_value)
     mark_to_change.save
   end
 
+  # Returns the criterion type.
+  def type
```

Comment:
You don't need to do this to convert `FlexibleCriterion` to `flexible_criteria`, Rails has helpers for automatically doing this sort of thing as long as the naming conventions are followed: http://api.rubyonrails.org/classes/ActiveSupport/Inflector.html

Look for `pluralize` and `underscore`